### PR TITLE
Switch to base36 instead of base64 for MD5 hashes to avoid using special characters

### LINKED ISSUE #119: MD5-based naming scheme breaks due to "/" in base64-encoded content
Thanks for creating this library! I'm trying to use this it to generate API documentation for my projects. I'm trying to use the `NameAndMd5Mix` mode to avoid long path issues I'm seeing with the default naming scheme.

The problem is that the MD5 hashes are encoded with base 64, which can contain the `/` character. This causes files to end up in nested folders (e.g. see [this file](https://github.com/madelson/DistributedLock/tree/default-documentation/docs/api/DistributedLock.Core)). This in turn breaks all relative links in the nested files (e.g. see the namespace link [here](https://github.com/madelson/DistributedLock/blob/default-documentation/docs/api/DistributedLock.Azure/AzureBlobLeaseDistributedLock..ctor.PPb%2B8c/p7LTz4wSm4vPWGg.md)).

I think an easy fix would be to use hex encoding rather than base 64. This has the added advantage of being case-insensitive which tends to be better for URLs.

If you're interested, I'd be happy to submit a PR.

### PR DESCRIPTION
**Pull request by madelson**
fix #119

### CODE REVIEW

File: source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs [hunk 1]
```diff
@@ -10,6 +10,6 @@ public sealed class Md5FactoryTest : AFileNameFactoryTest<Md5Factory>
         public void Name_Should_return_FullName() => Check.That(Name).IsEqualTo("Md5");
 
         [Fact]
-        public void GetFileName_Should_return_Md5() => Test(AssemblyInfo.ClassDocItem, "07ucmh7NagP7qlcO1w9snQ.md");
+        public void GetFileName_Should_return_Md5() => Test(AssemblyInfo.ClassDocItem, "NFMDH6WUUDQ1ZFSD9PKB7FIB9.md");
```
Comment (madelson) [inline source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs:13]: I think it would be nice to add a couple test cases here for the base36 encoding routine, especially around the logic to ensure non-negative and the logic to ensure we don't return an empty string.

However, this would require adding `[assembly: InternalsVisibleTo("DefaultDocumentation.Tests")]` to the DefaultDocumentation.Markdown project. Would that be OK?

Comment (Doraku) [reply source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs:13]: Sorry for the late reply! I am not a fan of `InternalsVisibleTo` but I get that you would want to secure the implementation with some tests. In essence the interface of IFileNameFactory should act almost as a passthrough anyway (the context is not really used and it will probably only use the `DocItem.FullName`), the `NamespaceDocItem` is probably the easiest implementation of `DocItem` to use to pass whatever you want. If it's really too cumbersome just use `InternalsVisibleTo` :).

Also sorry it seems the latest visual studio update added some analyzers that make the ci fails, I'll check to update the rules I want to keep!

Comment (madelson) [reply source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs:14]: @Doraku ok I've added a couple new tests using the mechanism you mentioned.

Comment (Doraku) [reply source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs:14]: I fixed the warning as error, if you rebase on master the ci should run the tests correctly now.

File: source/DefaultDocumentation.Markdown/FileNameFactories/Md5Factory.cs [hunk 2]
```diff
@@ -23,7 +24,42 @@ protected override string GetMarkdownFileName(IGeneralContext context, DocItem i
         {
             using MD5 md5 = MD5.Create();
```
[the remark this answers is not in the document]
Comment (madelson) [reply source/DefaultDocumentation.Markdown/FileNameFactories/Md5Factory.cs]: Oof good call.

File: source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs (hunk 1 above)
Comment (madelson) [reply source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs:14]: Thanks! Checks are passing now.

**Review by Doraku — APPROVED** [review]

**Comment by Doraku** [comment]
thanks again for working on this :)
